### ISSUE TITLE
[BUG/ISSUE] Error with flexgrid for custom domain

### ISSUE THREAD

**Issue by LukeFakes** [geoschem/geos-chem#403]
# Report a GEOS-Chem bug or technical issue

## Describe the bug:
I'm running Flexgrid at 0.5x0.625 for a custom domain. I've produced to 4x5 boundary conditions for that domain already. The model is using a blank 0.25x0.3125 restart file and 0.25x0.3125 met fields, which I think would be regraded by hemco? When running the model, I get a segmentation fault which I can't identify/understand the cause of. 

This doesn't occur when changing the only the defined resolution in input.geos to 0.25x03125 from (0.5x0.625) . For troubleshooting I've been running the model for a single day (start date 20181201 endnote 20181202). The error message makes me think it could be an issue in where the size of an output array is determined.

### Steps to reproduce: the bug:

1.  make realclean
2. make -j8 build BOUNDS=yes TRACEBACK=yes

### Error messages

```GET_BOUNDARY_CONDITIONS: Found All BCs at 2018/12/02 00:00
forrtl: severe (408): fort: (2): Subscript #3 of the array ARRAY has value 35 which is greater than the upper bound of 34

Image              PC                Routine            Line        Source             
libifcoremt.so.5   00002B9E0F0CD5A3  for_emit_diagnost     Unknown  Unknown
geos               00000000008FA4CB  diag3_                    757  diag3.F
geos               00000000006CF054  MAIN__                   1052  main.F
geos               000000000040397E  Unknown               Unknown  Unknown
libc-2.17.so       00002B9E10FBC495  __libc_start_main     Unknown  Unknown
geos               0000000000403899  Unknown               Unknown  Unknown
srun: error: node106: task 0: Exited with exit code 152
```

### Your GEOS-Chem version and runtime environment:

 - GEOS-Chem version:12.6
 - Compiler version: fort 18.0.3
 - netCDF version: netcdf4
 - Are you using GEOS-Chem "out of the box" (i.e. unmodified): yes

### Input and log files to attach

 - lastbuild: [lastbuild.txt](https://github.com/geoschem/geos-chem/files/5050377/lastbuild.txt)
 - input.geos: [input.geos.txt](https://github.com/geoschem/geos-chem/files/5050379/input.geos.txt)
 - HEMCO_Config.rc: [HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/5050362/HEMCO_Config.rc.txt)

 - GEOS-Chem "Classic" log file:[GEOS-Chem_Classic.geos.log.txt](https://github.com/geoschem/geos-chem/files/5050361/GEOS-Chem_Classic.geos.log.txt)

 - HEMCO.log: too large to upload

**Comment by yantosca** [comment]
Thanks for writing.  The error seems to be in this block in diag3.F:

```F90
      IF ( ND05 > 0 ) THEN
         CATEGORY = 'PL-SUL=$'

         DO M = 1, TMAX(5)
            N = TINDEX(5,M)

            ! Tracers 9, 10 are OH, NO3
            ! and are in [kg] instead of [kg S]
            IF ( N < 10 ) THEN
               UNIT = 'kg S'
            ELSE
               UNIT = 'kg'
            ENDIF

            NN     = N
            SCALEX = 1.e+0_fp

            DO L = 1, LD05
               ARRAY(:,:,L) = AD05(:,:,L,N) / SCALEX    ! <!--- line where error happens
            ENDDO

            CALL BPCH2( IU_BPCH,   MODELNAME, LONRES,   LATRES,
     &                  HALFPOLAR, CENTER180, CATEGORY, NN,
     &                  UNIT,      DIAGb,     DIAGe,    RESERVED,
     &                  State_Grid%NX, State_Grid%NY, LD05, IFIRST,
     &                  JFIRST,    LFIRST,    ARRAY(:,:,1:LD05) )
         ENDDO
      ENDIF
```

and I think I see why.  At the top of the routine we have:
```F90
      REAL(f4)           :: ARRAY(State_Grid%NX,State_Grid%NY,
     &                            State_Grid%NX+1)
```
but that should be:
```F90
      REAL(f4)           :: ARRAY(State_Grid%NX,State_Grid%NY,
     &                            State_Grid%NZ+1)
```
We will fix this in 13.0.0.  The ND05 diagnostics are now deprecated, as these can also be saved out via the HISTORY netCDF diagnostics.   We recommend using the netCDF diagnostic output whenever possible.

**Comment by yantosca** [comment]
Also note: the ND05 bpch diagnostics are removed in 13.0.0 as they are replaced by [the ProdLoss diagnostic collection](http://wiki.seas.harvard.edu/geos-chem/index.php/History_collections_for_chemistry_and_photolysis#The_ProdLoss_Collection).

**Comment by yantosca** [comment]
This issue is now resolved by commit 23dec8a31900bd1c900fa5b6de96158b262183ba.  This will be included in GEOS-Chem 13.0.0.
